packages/firestore/src/local/shared_client_state.ts

```diff
@@ -722,8 +731,8 @@ export class WebStorageSharedClientState implements SharedClientState {
     removedBatchIds: BatchId[],
     addedBatchIds: BatchId[]
   ): void {
```

Comment:
So it occurs to me that removedBatchIds is going to be the old mutation queue (for all clients) and addedBatchIds is going to be the new mutation queue (for all clients) and every client is going to run this handleUserChange() logic to add/remove the pending batches...  which probably isn't desirable and could potentially lead to race conditions...  Perhaps only the primary should do this or something?  Not 100% sure.

Reply:
Yeah, even if it is not racy, it is necessary to run this code more than once. I guarded it in a `isPrimary` check (in SyncEngine). Thanks for noticing.

Reply:
Hm, this breaks the Spec Tests and hence I just added a TODO for now.